src/looprestoration.rs

```diff
@@ -294,3 +298,106 @@ pub(crate) unsafe fn padding<BD: BitDepth>(
         }
     };
 }
+
+// TODO(randompoison): Temporarily public until init logic is deduplicated.
+pub(crate) unsafe extern "C" fn wiener_erased<BD: BitDepth>(
+    mut p: *mut libc::c_void,
+    stride: ptrdiff_t,
+    left: *const libc::c_void,
+    mut lpf: *const libc::c_void,
+    w: libc::c_int,
+    h: libc::c_int,
+    params: *const LooprestorationParams,
+    edges: LrEdgeFlags,
+    bitdepth_max: libc::c_int,
+) {
+    let bd = BD::new(bitdepth_max.as_());
```

Comment:
Can we instead add `fn from_c(bitdepth_max: c_int) -> Self` to `trait BitDepth`?  For `BitDepth8` this would just call `Self::new(())` and for `BitDepth16` it would call `Self::new(bitdepth_max.as_())`?  That way we keep a constant bitdepth for `BitDepth8`, as I just don't want that to accidentally interfere with any optimizations later on.

Reply:
Yup, that works too. Done!